src/Microsoft.Data.Sqlite/SqliteDataReader.cs

```diff
@@ -94,9 +95,17 @@ public override bool Read()
             }
 
             var rc = NativeMethods.sqlite3_step(_stmt);
-            MarshalEx.ThrowExceptionForRC(rc, _db);
+            try
+            {
+                MarshalEx.ThrowExceptionForRC(rc, _db);
+            }
+            catch
```

Comment:
I think we can rely on the contract of the class for this--that consumers must dispose. Their app has bigger problems if they're leaving readers open.